modules/system-info/reporters/server.php

```diff
@@ -305,34 +310,32 @@ public function get_mysql_version() {
 	 *                          folders don't have writing permissions, False otherwise.
 	 * }
 	 */
-	public function get_write_permissions() {
-		$paths_to_check = [
-			ABSPATH => 'WordPress root directory',
+	public function get_write_permissions() : array {
+		$paths_messages = [
+			static::KEY_PATH_WP_ROOT_DIR => 'WordPress root directory',
+			static::KEY_PATH_HTACCESS_FILE => '.htaccess file',
+			static::KEY_PATH_UPLOADS_DIR => 'WordPress uploads directory',
+			static::KEY_PATH_ELEMENTOR_UPLOADS_DIR => 'Elementor uploads directory',
 		];
 
-		$write_problems = [];
-
-		$wp_upload_dir = wp_upload_dir();
-
-		if ( $wp_upload_dir['error'] ) {
-			$write_problems[] = 'WordPress root uploads directory';
-		}
-
-		$elementor_uploads_path = $wp_upload_dir['basedir'] . '/elementor';
+		$paths_to_check = [
+			static::KEY_PATH_WP_ROOT_DIR => $this->get_system_path( static::KEY_PATH_WP_ROOT_DIR ),
+			static::KEY_PATH_HTACCESS_FILE => $this->get_system_path( static::KEY_PATH_HTACCESS_FILE ),
+			static::KEY_PATH_UPLOADS_DIR => $this->get_system_path( static::KEY_PATH_UPLOADS_DIR ),
+			static::KEY_PATH_ELEMENTOR_UPLOADS_DIR => $this->get_system_path( static::KEY_PATH_ELEMENTOR_UPLOADS_DIR ),
+		];
 
-		if ( is_dir( $elementor_uploads_path ) ) {
-			$paths_to_check[ $elementor_uploads_path ] = 'Elementor uploads directory';
-		}
+		$permissions = $this->get_paths_permissions( $paths_to_check );
 
-		$htaccess_file = ABSPATH . '/.htaccess';
+		$permissions = array_filter( $permissions, function ( $key ) use ( $paths_messages ) {
+			return array_key_exists( $key, $paths_messages );
+		}, ARRAY_FILTER_USE_KEY  );
```

Comment:
Do you still need it?

Reply:
Yes, it matches the path permissions to the proper message.

Reply:
Removed.